unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendsController.cs

```diff
@@ -36,26 +34,65 @@ public class FriendsController : IFriendsController
         public event Action<FriendRequest> OnFriendRequestReceived;
         public event Action<FriendRequest> OnSentFriendRequestApproved;
 
-        public static void CreateSharedInstance(IFriendsApiBridge apiBridge)
-        {
-            i = new FriendsController(apiBridge);
-        }
+        private CancellationTokenSource controllerCancellationTokenSource;
 
-        public FriendsController(IFriendsApiBridge apiBridge)
+        public FriendsController(IFriendsApiBridge apiBridge, IRPCSocialApiBridge rpcSocialApiBridge)
         {
             this.apiBridge = apiBridge;
-            apiBridge.OnInitialized += Initialize;
             apiBridge.OnFriendNotFound += FriendNotFound;
             apiBridge.OnFriendWithDirectMessagesAdded += AddFriendsWithDirectMessages;
             apiBridge.OnUserPresenceUpdated += UpdateUserPresence;
             apiBridge.OnFriendshipStatusUpdated += HandleUpdateFriendshipStatus;
             apiBridge.OnTotalFriendRequestCountUpdated += UpdateTotalFriendRequests;
             apiBridge.OnTotalFriendCountUpdated += UpdateTotalFriends;
-            apiBridge.OnFriendRequestsAdded += AddFriendRequests; // TODO (NEW FRIEND REQUESTS): remove when we don't need to keep the retro-compatibility with the old version
-            apiBridge.OnFriendRequestReceived += ReceiveFriendRequest;
+
+            this.socialApiBridge = rpcSocialApiBridge;
+
+            socialApiBridge.OnFriendAdded += AddFriend;
+            socialApiBridge.OnFriendRemoved += InternalRemoveFriend;
+            socialApiBridge.OnFriendRequestAdded += AddFriendRequest;
+            socialApiBridge.OnFriendRequestRemoved += RemoveFriendRequest;
         }
 
-        private void Initialize(FriendshipInitializationMessage msg)
+        public void Initialize()
+        {
+            controllerCancellationTokenSource = new CancellationTokenSource();
+
+            socialApiBridge.InitializeClient(controllerCancellationTokenSource.Token).Forget();
```

Comment:
perhaps its enough to call `apiBridge.Initialize()` and then your custom implementation of the social bridge do this internally. In the process you should trigger the initialization event so its propagated to the controller `InitializeFriendships` method

Reply:
I can prevent calling `InitializeSubscriptions` but this class awaits the result of `InitializeFriendshipsInformation` that's why it's a different method called from here

Reply:
Communicating via booleans and `await until` is expensive for a process, when it could just wait for the result of the task